R/zchunk_LA1321.cement.R

```diff
@@ -52,72 +63,369 @@ module_energy_LA1321.cement_DISABLED <- function(command, ...) {
     L132.in_EJ_R_indenergy_F_Yh <- get_data(all_data, "L132.in_EJ_R_indenergy_F_Yh")
 
     # ===================================================
-    # TRANSLATED PROCESSING CODE GOES HERE...
-    #
-    # If you find a mistake/thing to update in the old code and
-    # fixing it will change the output data, causing the tests to fail,
-    # (i) open an issue on GitHub, (ii) consult with colleagues, and
-    # then (iii) code a fix:
-    #
-    # if(OLD_DATA_SYSTEM_BEHAVIOR) {
-    #   ... code that replicates old, incorrect behavior
-    # } else {
-    #   ... new code with a fix
-    # }
-    #
-    #
-    # NOTE: there are 'match' calls in this code. You probably want to use left_join_error_no_match
-    # For more information, see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # NOTE: This code uses vecpaste
-    # This function can be removed; see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # ===================================================
+    # 2. Perform computations
+
+    # Set constants used for this chunk
+    # ---------------------------------
+
+    # Extract carbon coefficient for limestone from assumption file
+    LIMESTONE_CCOEF <- A_PrimaryFuelCCoef$PrimaryFuelCO2Coef[A_PrimaryFuelCCoef$PrimaryFuelCO2Coef.name == "limestone"]
+    # Determine historical years not available in data set (additional years) to be copy values from final available year (final_CO2_year)
+    ADDITIONAL_YEARS <- HISTORICAL_YEARS[!HISTORICAL_YEARS %in% energy.CDIAC_CO2_HISTORICAL_YEARS]
+    FINAL_CO2_YEAR <- energy.CDIAC_CO2_HISTORICAL_YEARS[length(energy.CDIAC_CO2_HISTORICAL_YEARS)]
+    # Sets maximum for IO coefficient calculated below
+    MAX_IOELEC <- 4
+
+    # =======================================================================================
+    # Derivation of cement production and limestone consumption by region and historical year
+    # =======================================================================================
+
+    # Downscale Worrell's process CO2 emissions and cement production in 1994 to country level using CDIAC emissions inventory
+    # ------------------------------------------------------------------------------------------------------------------------
+    cement_regions %>%
+      left_join(L100.CDIAC_CO2_ctry_hist, by = "iso") %>%
+      filter(year == 1994) %>%
```

Comment:
?

Reply:
Worrell data is only for 1994, so those ratios are assumed to be constant for other years. You have to filter CDIAC to the 1994 values to match it. It's not great, but it's the way it's set up.

Reply:
Can you add an explanatory comment about this then?

Reply:
yup, already done! (this time without clicking the wrong button and triggering a review)